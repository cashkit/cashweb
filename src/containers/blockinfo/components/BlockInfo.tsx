import React from 'react';

export interface BlockInfoProps {
    height: number,
    version: number,
    timestamp: number,
    bits: number,
    nonce: number,
    confirmations: number,
    difficulty: number,
    size: number,
    medianTime: number,
    transactions: number | false
}

export const BlockInfo = ({ height, version, timestamp, bits, nonce,
    confirmations, difficulty, size, medianTime, transactions }
    :  BlockInfoProps) => {
  return(
    <>
      <div className="tile is-ancestor">
        <div className="tile is-parent">
          <article className="tile is-child box has-text-left">
            <p className="is-size-4 has-text-weight-medium">Height</p>
            <div className="content">{height}</div>
          </article>
        </div>
        <div className="tile is-parent">
          <article className="tile is-child box has-text-left">
            <p className="is-size-4 has-text-weight-medium">Version</p>
            <div className="content">{version}</div>
          </article>
        </div>
        <div className="tile is-parent">
          <article className="tile is-child box has-text-left">
            <p className="is-size-4 has-text-weight-medium">Difficulty</p>
            <div className="content">{difficulty}</div>
          </article>
        </div>
        <div className="tile is-parent">
          <article className="tile is-child box has-text-left">
            <p className="is-size-4 has-text-weight-medium">Timestamp</p>
            <div className="content">{timestamp}</div>
          </article>
        </div>
        <div className="tile is-parent">
          <article className="tile is-child box has-text-left">
            <p className="is-size-4 has-text-weight-medium">Bits</p>
            <div className="content">{bits}</div>
          </article>
        </div>
      </div>
  
      <div className="tile is-ancestor">
        <div className="tile is-parent">
          <article className="tile is-child box has-text-left">
            <p className="is-size-4 has-text-weight-medium">Nonce</p>
            <div className="content">{nonce}</div>
          </article>
        </div>
        <div className="tile is-parent">
          <article className="tile is-child box has-text-left">
            <p className="is-size-4 has-text-weight-medium">Confirmations</p>
            <div className="content">{confirmations}</div>
          </article>
        </div>
  
        <div className="tile is-parent">
          <article className="tile is-child box has-text-left">
            <p className="is-size-4 has-text-weight-medium">Size</p>
            <div className="content">{size}</div>
          </article>
        </div>
        <div className="tile is-parent">
          <article className="tile is-child box has-text-left">
            <p className="is-size-4 has-text-weight-medium">Median Time</p>
            <div className="content">{medianTime}</div>
          </article>
        </div>
        <div className="tile is-parent">
          <article className="tile is-child box has-text-left">
            <p className="is-size-4 has-text-weight-medium">Total Transactions</p>
            <div className="content">{transactions}</div>
          </article>
        </div>
      </div>
      </>
    )
}
