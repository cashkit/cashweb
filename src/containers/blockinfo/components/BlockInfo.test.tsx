import React from 'react';
import renderer from 'react-test-renderer';

import { BlockInfo, BlockInfoProps } from './BlockInfo';

const props: BlockInfoProps = {
  height: 0,
  version: 0,
  timestamp: 0,
  bits: 0,
  nonce: 0,
  confirmations: 0,
  difficulty: 0,
  size: 0,
  medianTime: 0,
  transactions: 0,
}

it('<BlockInfo/> renders correctly', () => {
  const tree = renderer
    .create(<BlockInfo {...props}></BlockInfo>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});