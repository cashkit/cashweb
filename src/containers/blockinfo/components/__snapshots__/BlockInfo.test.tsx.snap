// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlockInfo/> renders correctly 1`] = `
Array [
  <div
    className="tile is-ancestor"
  >
    <div
      className="tile is-parent"
    >
      <article
        className="tile is-child box has-text-left"
      >
        <p
          className="is-size-4 has-text-weight-medium"
        >
          Height
        </p>
        <div
          className="content"
        >
          0
        </div>
      </article>
    </div>
    <div
      className="tile is-parent"
    >
      <article
        className="tile is-child box has-text-left"
      >
        <p
          className="is-size-4 has-text-weight-medium"
        >
          Version
        </p>
        <div
          className="content"
        >
          0
        </div>
      </article>
    </div>
    <div
      className="tile is-parent"
    >
      <article
        className="tile is-child box has-text-left"
      >
        <p
          className="is-size-4 has-text-weight-medium"
        >
          Difficulty
        </p>
        <div
          className="content"
        >
          0
        </div>
      </article>
    </div>
    <div
      className="tile is-parent"
    >
      <article
        className="tile is-child box has-text-left"
      >
        <p
          className="is-size-4 has-text-weight-medium"
        >
          Timestamp
        </p>
        <div
          className="content"
        >
          0
        </div>
      </article>
    </div>
    <div
      className="tile is-parent"
    >
      <article
        className="tile is-child box has-text-left"
      >
        <p
          className="is-size-4 has-text-weight-medium"
        >
          Bits
        </p>
        <div
          className="content"
        >
          0
        </div>
      </article>
    </div>
  </div>,
  <div
    className="tile is-ancestor"
  >
    <div
      className="tile is-parent"
    >
      <article
        className="tile is-child box has-text-left"
      >
        <p
          className="is-size-4 has-text-weight-medium"
        >
          Nonce
        </p>
        <div
          className="content"
        >
          0
        </div>
      </article>
    </div>
    <div
      className="tile is-parent"
    >
      <article
        className="tile is-child box has-text-left"
      >
        <p
          className="is-size-4 has-text-weight-medium"
        >
          Confirmations
        </p>
        <div
          className="content"
        >
          0
        </div>
      </article>
    </div>
    <div
      className="tile is-parent"
    >
      <article
        className="tile is-child box has-text-left"
      >
        <p
          className="is-size-4 has-text-weight-medium"
        >
          Size
        </p>
        <div
          className="content"
        >
          0
        </div>
      </article>
    </div>
    <div
      className="tile is-parent"
    >
      <article
        className="tile is-child box has-text-left"
      >
        <p
          className="is-size-4 has-text-weight-medium"
        >
          Median Time
        </p>
        <div
          className="content"
        >
          0
        </div>
      </article>
    </div>
    <div
      className="tile is-parent"
    >
      <article
        className="tile is-child box has-text-left"
      >
        <p
          className="is-size-4 has-text-weight-medium"
        >
          Total Transactions
        </p>
        <div
          className="content"
        >
          0
        </div>
      </article>
    </div>
  </div>,
]
`;
